# This file is part of HiGFXback

# requires
REQUIRES="lua-build make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=

# build
make

# install
install -d $DESTDIR/lib/lua/5.1
install src/lfs.so $DESTDIR/lib/lua/5.1

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/luafilesystem-build.pc << EOF
Name: luafilesystem
Version: 1.8.0
Description: File system library for Lua
Requires: $REQUIRES

exec=\\
/lib/lua/5.1/lfs.so
EOF

exit

# patch
--- luafilesystem-1.8.0.orig/Configure
+++ luafilesystem-1.8.0/Configure
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|=$PREFIX|" config
--- luafilesystem-1.8.0.orig/Makefile
+++ luafilesystem-1.8.0/Makefile
@@ -12,7 +12,7 @@
 lib: src/lfs.so
 
 src/lfs.so: $(OBJS)
-	MACOSX_DEPLOYMENT_TARGET=$(MACOSX_DEPLOYMENT_TARGET); export MACOSX_DEPLOYMENT_TARGET; $(CC) $(LIB_OPTION) -o src/lfs.so $(OBJS)
+	$(CC) $(LIB_OPTION) -o src/lfs.so $(OBJS)
 
 test: lib
 	LUA_CPATH=./src/?.so lua tests/test.lua
--- luafilesystem-1.8.0.orig/config
+++ luafilesystem-1.8.0/config
@@ -8,21 +8,11 @@
 # System's libraries directory (where binary libraries are installed)
 LUA_LIBDIR= $(PREFIX)/lib/lua/$(LUA_VERSION)
 
-# Lua includes directory
-LUA_INC += -I$(PREFIX)/include
-LUA_INC += -I/usr/include/lua$(LUA_VERSION) -I/usr/include/lua/$(LUA_VERSION)
-
 # OS dependent
 LIB_OPTION= -shared #for Linux
-#LIB_OPTION= -bundle -undefined dynamic_lookup #for MacOS X
-
-# Minimum runtime OS version on macOS
-MACOSX_DEPLOYMENT_TARGET= 10.5
 
 # Compilation directives
-WARN= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
-INCS= $(LUA_INC)
-CFLAGS= $(WARN) $(INCS)
+CFLAGS= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
 CC= gcc
 
 # $Id: config,v 1.21 2007/10/27 22:42:32 carregal Exp $

# source
https://github.com/lunarmodules/luafilesystem/archive/v1_8_0/luafilesystem-1.8.0.tar.gz
